File: AccountintApp/app/src/main/java/edu/coloradomesa/cs/accountintapp/MainActivity.java
package edu.coloradomesa.cs.accountintapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private ArrayList<Account> Accounts = new ArrayList<Account>();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ArrayList<Account> tempAccounts = getIntent().getParcelableArrayListExtra("accountsList");
        if(tempAccounts != null) Accounts = tempAccounts;
    }

    public void addAccount(View view) {
        String acct, val, type;
        TextView tAcct = (TextView) findViewById(R.id.name_on_account);
        TextView tVal = (TextView) findViewById(R.id.amount_start);
        Spinner spinner = (Spinner)findViewById(R.id.type_account);
        type = spinner.getSelectedItem().toString();
        acct = tAcct.getText().toString();
        val = tVal.getText().toString();
        TextView error = (TextView) findViewById(R.id.error_message);
        if(acct.equals("") || val.equals("")) {
            error.setText(R.string.error_mssg);
        }
        else {
            error.setText(acct + " added");
            Account knew = new Account(acct,type,Integer.parseInt(val));
            Accounts.add(knew);
            tAcct.setText("");
            tVal.setText("");
        }
    }

    public void toTransactionPage(View view) {
        Intent i = new Intent(this, TransactionActivity.class);
        i.putParcelableArrayListExtra("accountsList",Accounts);
        startActivity(i);
    }

    public void toStatementPage(View view) {
        Intent i = new Intent(this, StatementActivity.class);
        i.putParcelableArrayListExtra("accountsList",Accounts);
        startActivity(i);
    }
}

File: ClimbingDocumentation/app/src/main/java/edu/coloradomesa/cs/climbingdocumentation/MainActivity.kt
package edu.coloradomesa.cs.climbingdocumentation

import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private var climbsList = ArrayList<ClimbInfo>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val climbsListTemp = intent.getSerializableExtra("climbsList") as? ArrayList<ClimbInfo>
        if (climbsListTemp != null) {
            climbsList = climbsListTemp
        }
    }


    fun goToDisplayList(view: View) {
        val intent = Intent(this, DisplayActivity::class.java)
        intent.putExtra("climbsList",climbsList)
        startActivity(intent)
    }

    fun addClimbToList(view: View) {
        var add: Boolean = true
        if (mainClimbName.text.isEmpty()) {
            mainClimbName.setText(R.string.no_name)
            add = false
        }
        if (mainClimbGrade.text.isEmpty()) {
            mainClimbGrade.setText(R.string.no_grade)
            add = false
        }
        if (mainClimbDate.text.isEmpty()) {
            mainClimbDate.setText(R.string.no_date)
            add = false
        }
        if(add) {
            val type: String
            if(boulderClimb.isChecked) type = "Boulder"
            else type = "Roped Climb"
            val newClimb = ClimbInfo(type, mainClimbName.text.toString(), mainClimbGrade.text.toString(), mainClimbDate.text.toString())
            climbsList.add(newClimb)
            mainClimbName.setText("")
            mainClimbGrade.setText("")
            mainClimbDate.setText("")
        }
    }
}
